### PR TITLE
Jenkinsfile guidelines

### PR DESCRIPTION
**Pull request by nie7321**
## Overview

All things `Jenkinsfile`.

@cecolby11 @bab4379 can you think of anything else we should cover?

## Checklist

- [x] Proof-reading
- [x] Any diagrams/figures have their source files included

### CODE REVIEW

File: ci-cd/jenkins.md [hunk 1]
```diff
@@ -1 +1,144 @@
...
+
+We typically keep pipeline files in a `.jenkins/` folder. Even if you only have a single Jenkinsfile, it is advisable to put it in the folder; changing it later is a nuisance. 
+
+## Configuration
+As much configuration as possible should be included in the Jenkinsfile. 
+
+- Review the [options](https://jenkins.io/doc/book/pipeline/syntax/#options) section
+    - `disableConcurrentBuilds()` is a commonly-used option; you wouldn't want to `terraform apply` off the same tfstate file at the same time.
+- Review the [triggers](https://jenkins.io/doc/book/pipeline/syntax/#triggers) section
+    - `pollSCM('H/10 * * * *')` is popular; Jenkins is firewalled off so Github cannot notify it when code is pushed
```
Comment (cecolby11) [inline ci-cd/jenkins.md]: what is our official stance on prod/master polling? I feel like up to this point we've discouraged polling prod, but maybe we are of the mind that anything you put in prod you better be ok with deploying? or do we want to encourage people tagging releases in prod, in which case I'm not sure how useful polling would be.

File: ci-cd/jenkins.md [hunk 2]
```diff
@@ -1 +1,144 @@
...
+            office365ConnectorSend status: "Failed", webhookUrl: "${env.TEAMS_WEBHOOK_URL}"
+        }
+    }
+}
+```
+
+## Scheduling Deployments
+You must regularly re-deploy your application. Our minimum is once per week, but more frequent deployments are encouraged.
+
+Re-deploying on a schedule, even when no changes are pending deployment, ensures that your pipeline & deployment process valid. Some applications may go into a period of development inactivity; it's good to know that you can apply a critical security patch without worrying if your pipeline still works after not being used for nine months.
```
Comment (bab4379) [inline ci-cd/jenkins.md]: maybe missing a word "are/is/remains" between process and valid

File: ci-cd/jenkins.md (hunk 1 above)
Comment (bab4379) [reply ci-cd/jenkins.md]: Well I accidentally looked at my mouse wrong and ended up going back and lost my comment.  Basically I do not have a strong opinion on polling of prod.  I can make arguments for and against it.  If we needed to take a stance, based on the way NU does things, like CAB meetings and only off hours deployments, then I would say do not poll prod and do automated deploys.

Comment (nie7321) [reply ci-cd/jenkins.md]: I've put my notes on this together and set up a discussion for next week.

This article isn't directly dealing with that question -- is it OK to merge?

**Pull request merged**